lib/core/src/sdk.rs

```diff
@@ -1976,6 +1976,24 @@ impl LiquidSdk {
         Ok(self.persister.get_payments(req)?)
     }
 
+    /// Retrieves a payment by its destination.
+    ///
+    /// # Arguments
+    ///
+    /// * `destination` - The destination of the payment to retrieve.
+    ///
+    /// # Returns
+    ///
+    /// Returns an `Option<Payment>` if found, or `None` if no payment matches the given destination.
+    pub async fn payment_by_destination(
+        &self,
+        destination: &PaymentDestination,
+    ) -> Result<Option<Payment>, PaymentError> {
+        self.ensure_is_started().await?;
+
+        Ok(self.persister.get_payment_by_destination(destination)?)
+    }
```

Comment:
Looking back on the changes, wouldn't it be better to name this `get_payment_by_destination` like the persister method? I'm afraid this could be misinterpreted as executing a payment by destination.

Reply:
Agreed, I've since added `payment_hash` as a query option so `get_payment()` makes more sense now